client/app/scripts/superdesk-dashboard/module.js

```diff
@@ -38,26 +33,19 @@ define([
                     }
                 };
 
-                desks.fetchCurrentUserDesks()
-                    .then(function(userDesks) {
-                        scope.userDesks = userDesks._items;
-
-                        var currentDeskId = desks.getCurrentDeskId();
-                        if (!currentDeskId) {
-                            preferencesService.get('desk:last_worked').then(
-                                function(desk) {
-                                    if (desk !== '' && desk !== 'personal') {
-                                        scope.select(_.find(scope.userDesks, {_id: desk}), false);
-                                    } else {
-                                        scope.select(scope.userDesks[0], false);
-                                    }
-                            },  function() {
-                                    scope.select(scope.userDesks[0], false);
-                            });
-                        } else {
-                            scope.select(_.find(scope.userDesks, {_id: currentDeskId}), false);
-                        }
+                desks.initialize()
+                .then(function() {
+                    desks.fetchCurrentUserDesks().then(function (userDesks) {
+                    	scope.userDesks = userDesks._items;
+                    	if (!desks.activeDeskId && scope.userDesks) {
+                    		scope.select(scope.userDesks[0], false);
+                    	} else {
+                    		if (desks.getCurrentDesk() !== scope.selected) {
```

Comment:
here it might be more obvious to test for `!= null`, `scope.selected` is not yet defined anyhow right?